daal4py/sklearn/ensemble/tests/test_decision_forest.py

```diff
@@ -0,0 +1,173 @@
+    },
+    {
+        0: random.uniform(1, 10),
+        1: random.uniform(50, 100),
+        2: random.uniform(1, 100),
+    },
+    {
+        0: random.uniform(1, 10),
+        1: random.uniform(1, 100),
+        2: random.uniform(1, 1000),
+    },
+    {
+        0: random.uniform(1, 2000),
+        1: random.uniform(1, 2000),
+        2: random.uniform(1, 2000),
+    },
+    {0: 50, 1: 50, 2: 50},
+    'balanced',
+]
+
+
+@pytest.mark.parametrize('weight', CLASS_WEIGHTS_IRIS)
+def test_classifier_class_weight_iris(weight):
+    if daal_check_version((2020, 3)):
+        check_classifier_class_weight_iris(weight)
+
+
+def check_classifier_class_weight_iris(weight, check_ratio=0.85):
+    alg_results = []
+    for _ in range(N_TRIES):
+        X, y = load_iris(return_X_y=True)
+        X_train, x_test, y_train, y_test = \
+            train_test_split(X, y, test_size=0.33, random_state=31)
+
+        SK_model = SKRandomForestClassifier(class_weight=weight)
+        D4P_model = D4PRandomForestClassifier(class_weight=weight)
+
+        SK_predict = SK_model.fit(X_train, y_train).predict(x_test)
+        D4P_predict = D4P_model.fit(X_train, y_train).predict(x_test)
+
+        SK_accuracy = accuracy_score(SK_predict, y_test)
+        D4P_accuracy = accuracy_score(D4P_predict, y_test)
+        ratio = D4P_accuracy / SK_accuracy
+        alg_results.append(ratio)
+
+    assert np.mean(alg_results) >= check_ratio
+
+SAMPLE_WEIGHTS_IRIS = [
+    (np.full_like(range(100), 0), 'Only 0'),
+    (np.full_like(range(100), 1), 'Only 1'),
+    (np.full_like(range(100), 5), 'Only 5'),
+    (np.full_like(range(100), 50), 'Only 50'),
+    (np.random.rand(100), 'Uniform distribution'),
+    (np.random.normal(1000, 10, 100), 'Gaussian distribution'),
+    (np.random.exponential(5, 100), 'Exponential distribution'),
+    (np.random.poisson(lam=10, size=100), 'Poisson distribution'),
+    (np.random.rayleigh(scale=1, size=100), 'Rayleigh distribution'),
+]
+
+
+@pytest.mark.parametrize('weight', SAMPLE_WEIGHTS_IRIS)
+def test_classifier_sample_weight_iris(weight):
+    if daal_check_version((2020, 3)):
+        check_classifier_sample_weight(weight)
+
+
+def check_classifier_sample_weight(weight, check_ratio=0.9):
+    alg_results = []
+    for _ in range(N_TRIES):
+        X, y = load_iris(return_X_y=True)
+        X_train, x_test, y_train, y_test = \
+            train_test_split(X, y, test_size=0.33, random_state=31)
+
+        SK_model = SKRandomForestClassifier()
+        D4P_model = D4PRandomForestClassifier()
+
+        SK_predict = SK_model.fit(X_train, y_train,
+                                  sample_weight=weight[0]).predict(x_test)
+        D4P_predict = D4P_model.fit(X_train, y_train,
+                                    sample_weight=weight[0]).predict(x_test)
+
+        SK_accuracy = accuracy_score(SK_predict, y_test)
+        D4P_accuracy = accuracy_score(D4P_predict, y_test)
+        ratio = D4P_accuracy / SK_accuracy
```

Comment:
I suppose we need to check not only accuracy here by predicted labels, but additionally AUC/logloss by predicted probabilities. And probably, other metrics like F1, precision, ...